src/Discovery/src/Eureka/EurekaPostConfigurer.cs

```diff
@@ -191,6 +182,37 @@ public static void UpdateConfiguration(IConfiguration config, EurekaServiceInfo
         }
     }
 
+    private static void AssertValid(EurekaServiceInfo serviceInfo, EurekaClientOptions clientOptions)
+    {
+        if (!clientOptions.Enabled)
+        {
+            return;
+        }
+
+        if (!Platform.IsContainerized && !Platform.IsCloudHosted)
+        {
+            return;
+        }
+
+        if (serviceInfo != null)
+        {
+            return;
+        }
+
+        if (!clientOptions.EurekaServerServiceUrls.Contains(EurekaClientConfig.DefaultServerServiceUrl.TrimEnd('/')))
+        {
+            return;
+        }
+
+        if (!clientOptions.ShouldRegisterWithEureka && !clientOptions.ShouldFetchRegistry)
+        {
+            return;
+        }
+
+        throw new InvalidOperationException(
```

Comment:
Possibly outside the scope of this PR, but if we can log instead of throwing it _might_ be preferable https://github.com/SteeltoeOSS/Steeltoe/issues/829

Reply:
The linked issue is about `ConsulPostConfigurer` (called from `ValidateConsulOptions()`), while this comment is on `EurekaPostConfigurer` (called from `UpdateConfiguration()`). Does the same concern apply here?

For both, we could add `bool AreOptionsValid()`, which enables callers to check upfront. Then the callers can be expanded with a `bool throwOnError` parameter. These are just suggestions without me fully understanding the context, so my thinking is this is probably better first discussed in the linked issue, followed by a separate PR (and possibly backported).

Reply:
Yes the same concern applies. yeah, let's postpone changes for now